sync/src/main/java/org/bf2/sync/SecretManager.java

```diff
@@ -108,67 +100,67 @@ private Secret buildSecret(String name, String type, ManagedKafka managedKafka,
+        final String masterSecretName = masterSecretName(managedKafka);
+        final ManagedKafkaSpecBuilder replacement = new ManagedKafkaSpecBuilder(managedKafka.getSpec());
+
+        if (isKafkaExternalCertificateEnabled(managedKafka)) {
+            replacement.editEndpoint()
+                .withNewTls()
+                    .withNewCertRef(ENDPOINT_TLS_CRT, masterSecretName, null)
+                    .withNewKeyRef(ENDPOINT_TLS_KEY, masterSecretName, null)
+                .endTls()
+                .endEndpoint();
         }
-        if(isCanaryServiceAccountPresent(managedKafka)){
-            ServiceAccount serviceAccount = managedKafka.getServiceAccount(ServiceAccount.ServiceAccountName.Canary).get();
-            serviceAccount.setPrincipal(null);
-            serviceAccount.setPassword(null);
-            serviceAccount.setPrincipalRef(new SecretKeySelectorBuilder()
-                    .withName(secret.getMetadata().getName())
-                    .withKey(CANARY_SASL_PRINCIPAL)
-                    .build());
-            serviceAccount.setPasswordRef(new SecretKeySelectorBuilder()
-                    .withName(secret.getMetadata().getName())
-                    .withKey(CANARY_SASL_PASSWORD)
-                    .build());
+
+        if (isKafkaAuthenticationEnabled(managedKafka)) {
+            replacement.editOauth()
+                    .withNewClientIdRef(OAUTH_SSO_CLIENT_ID, masterSecretName, null)
+                    .withClientId(null)
+                    .withNewClientSecretRef(OAUTH_SSO_CLIENT_SECRET, masterSecretName, null)
+                    .withClientSecret(null)
+                .endOauth();
         }
-       managedKafka = new ManagedKafkaBuilder(managedKafka)
-                .editOrNewMetadata()
-                    .addToAnnotations(Map.of(ANNOTATION_MASTER_SECRET_DIGEST, buildDigest(decode(secret.getData()))))
-                .endMetadata()
+
+        managedKafka.getServiceAccount(ServiceAccount.ServiceAccountName.Canary)
+            .ifPresent(serviceAccount -> {
+                replacement.editMatchingServiceAccount(sa -> serviceAccount.getName().equals(sa.getName()))
+                    .withNewPrincipalRef(CANARY_SASL_PRINCIPAL, masterSecretName, null)
+                    .withPrincipal(null)
+                    .withNewPasswordRef(CANARY_SASL_PASSWORD, masterSecretName, null)
+                    .withPassword(null)
+                .endServiceAccount();
+            });
+
+       return new ManagedKafkaBuilder(managedKafka)
+                .withSpec(replacement.build())
                 .build();
-        return managedKafka;
+    }
+
+    public void calculateMasterSecretDigest(ManagedKafka managedKafka, Secret secret) {
+        var metadata = new ObjectMetaBuilder(managedKafka.getMetadata())
+                .addToAnnotations(ANNOTATION_MASTER_SECRET_DIGEST, buildDigest(decode(secret.getData())))
+                .build();
+
+        managedKafka.setMetadata(metadata);
```

Comment:
@MikeEdgar Could you help with a little bit more information about why we creating this `calculateMasterSecretDigest` method and then building the digest?